File: processcontrol.client/src/app/models/RamStagesPerIv.ts
import { RamInvoices } from "./RamInvoices";
import { RamIvStages } from "./RamIvStages";

export class RamStagesPerIv {
  stageIv!: number;
  invoiceIdf!: number;
  stageIdf!: number;
  resStatus!: string;
  uDate!: Date;
  errDes!: string;
  ivRequest!: string;
  stageDes!: string;

  stageId!: RamIvStages[];
  invoiceId!: RamInvoices[];
}

File: ProcessControl.Server/Models/Ramdor/RamStagesPerIv.cs
using System.ComponentModel.DataAnnotations;

namespace ProcessControl.Server.Models.Ramdor
{
    public class RamStagesPerIv
    {
        [Key]
        public int StageIv { get; set; }
        public int InvoiceIdf {  get; set; }
        public int StageIdf { get; set; }
        public string? ResStatus { get; set; }
        public DateTime? UDate { get; set; }
        public string? ErrDes { get; set; }
        public string? IvRequest { get; set; }
        public virtual RamIvStages StageId { get; set; }
        public virtual RamInvoices InvoiceId { get; set; }
        public RamStagesPerIv() { }
    }
}
